version: "3.8"

services:
  postgres:
    image: postgres:16-alpine
    environment:
      POSTGRES_DB: firmable
      POSTGRES_USER: admin
      POSTGRES_PASSWORD: admin
    ports:
      - "5432:5432"
    volumes:
      - pgdata:/var/lib/postgresql/data
      - ./init.sql:/docker-entrypoint-initdb.d/init.sql:ro
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U admin"]
      interval: 5s
      timeout: 5s
      retries: 5

  pgadmin:
    image: dpage/pgadmin4
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@firmable.com
      PGADMIN_DEFAULT_PASSWORD: admin
    ports:
      - "5050:80"
    depends_on:
      - postgres

  etl:
    build: .
    volumes:
      - ./data:/app/data
      - ./dbt:/app/dbt
    environment:
      DATABASE_URL: postgresql+psycopg2://admin:admin@postgres:5432/firmable
      DBT_PROFILES_DIR: /app/dbt
      DBT_PROJECT_DIR: /app/dbt/prrprocess
      DBT_HOST: postgres
      DBT_USER: admin
      DBT_PASSWORD: admin
      DBT_PORT: 5432
      DBT_DB: firmable
      DBT_SCHEMA: public
    depends_on:
      postgres:
        condition: service_healthy
    stdin_open: true
    tty: true
    command: >
          python run.py --run-dbt --dbt-path ./dbt/prrprocess --entity-matching
volumes:
  pgdata:
